src/shmem/team.c

```diff
@@ -17,7 +17,9 @@ int shmem_team_my_pe(shmem_team_t team)
         return -1;
     }
 
+    OSHMPI_THREAD_ENTER_CS(&(_team->cs));
     _team = OSHMPI_TEAM_HANDLE_TO_OBJ(team);
+    OSHMPI_THREAD_EXIT_CS(&(_team->cs));
     return _team->my_pe;
```

Comment:
I was thinking about the case when one thread uses the team while other thread tries to destroy the team. I realized this commit does not provide that. I am going to remove this one commit and revisit this problem later when working on collectives.

Reply:
I don't think the above case is allowed by the spec. See **9.4 Team Management Routines** at page 31 of spec 1.5

```
Thread Safety
When it is allowed by the threading model provided by the OpenSHMEM library, a team may be used concurrently in 
non-collective operations (e.g., shmem_team_my_pe) by multiple threads within the PE where it was created. A team 
may not be used concurrently by multiple threads in the same PE for collective operations. However, multiple collective 
operations on different teams may be performed in parallel.
```

In addition, in **9.9 Collective Routines** at page 88 of spec 1.5, it also states that:
```
Team-based collectives
...
While OpenSHMEM routines provide thread support according to the thread-support level provided at initialization 
(see Section 9.2), team-based collective routines may not be called simultaneously by multiple threads on a given
team.
```
I believe the intention is to avoid critical section at collective fast-path as much as possible. For cases where the user wants to overlap collectives with multithreads, they should create separate teams in advance.

Reply:
Understood. I have removed the lock now.